### ISSUE TITLE
Test if Tradfri light is available/reachable?

### ISSUE THREAD

**Issue by Mariusthvdb** [ggravlingen/pytradfri#196]
### Where are you using pytradfri (eg stand-alone, Home Assistant etc)
Homeassistant
### Expected behaviour
Please consider adding functionality of checking reachability of Tradfri-lights. 

### Actual behaviour
Right now, when power is cut off to the lights, they still show on or off. In the app the lights show unavailable (grey out) so some sort of check is already implemented in the Tradfri system.

please see community-post: 
https://community.home-assistant.io/t/test-if-tradfri-light-is-available-reachable/74314

if reachability would be implemented we can do this in the Ha frontend, and, create automations and other template sensors based on that reachabilty:

![schermafbeelding 2018-10-21 om 14 11 32](https://user-images.githubusercontent.com/33354141/47266771-62ab9980-d53b-11e8-8f80-b35e8c137fa1.jpg)

![schermafbeelding 2018-10-21 om 14 16 58](https://user-images.githubusercontent.com/33354141/47266826-0dbc5300-d53c-11e8-9976-82549c54def0.jpg)
![schermafbeelding 2018-10-21 om 14 17 10](https://user-images.githubusercontent.com/33354141/47266827-0e54e980-d53c-11e8-8a4a-dccbc6faacb9.jpg)

Driveway is a Philips Hue light tested with:

```
- platform: command_line
  name: Driveway reachable
  command: !secret driveway_command
  value_template: >
    {{value_json.state.reachable}}
```

Parking is a Tradfri light showing to be on, while in fact it is unavailable while power is cut.

**Comment by ggravlingen** [comment]
I did some investigation here and in the code base of Home Assistant and it seems (which I had forgotten) the feature is already available. In pytradfri, the reachable state is exposed here:
https://github.com/ggravlingen/pytradfri/blob/3e3d68fb5b14c497b45548085778261fbd3c0ff7/pytradfri/device.py#L60

And in Home Assisstant, here:
https://github.com/home-assistant/home-assistant/blob/7f0a50ce31befab10e8ba1892cc30b1334f22fc2/homeassistant/components/light/tradfri.py#L359

I also tried switching of my light using the wall switch and the light does show up as unavailable:
![image](https://user-images.githubusercontent.com/21142447/47269054-fcce0a80-d558-11e8-82ee-8000aeeb6891.png)

So, in conclusion. It already does work :)

**Comment by Mariusthvdb** [comment]
Not sure how you did this , but it is exactly what I am hoping for .
If power is cut off here , the lights show with their last state, I haven’t seen unavailable on any of the TRÅDFRI lights yet.

**Comment by ggravlingen** [comment]
Just to be sure: you are running the tradfri gateway and not a hue?

**Comment by Mariusthvdb** [comment]
Yes I am.
Both in fact ;-) 
But I use the dedicated setup: TRÅDFRI lights on the TRÅDFRI hub and hue lights on the hue hub.

**Comment by ggravlingen** [comment]
I'm running v80.3 of Home Assistant, which runs the latest version of pytradfri. But there have not been any changes to this part of the code for long. What happens if you turn of a light using the switch and then try to adjust, for instance, brightness?

**Comment by Mariusthvdb** [comment]
a yes, now i see the too: unavailable. Neer noticed before. Thats a start!
It needs the command though and doesn't change to unavailable automatically when power is cut.

**Comment by ggravlingen** [comment]
That's something that needs to be fixed in home assistant so we should close the issue here.

(I'm not enough versed in the inner workings of home assistant to provide a fix in home assistant I'm afraid.)

**Comment by Mariusthvdb** [comment]
ok, where should we go from here then? the regular homeassistant GitHub issue tracker?

**Comment by ggravlingen** [comment]
Yes and hope that someone picks the issue up. As a temporary workaround, perhaps you could trigger an automation that sends the last observed brightness level every five minutes or so? Just to trigger the unavailable state.

**Comment by ggravlingen** [comment]
This issue does not relate to pytradfri, but rather the Home Assistant project. Closing the issue so that the discussion can be continued in their repo.
